dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -380,61 +380,109 @@ private void logFullWorkflowRefresh(final Workflow workflow) {
         }
     }
 
+    /**
+     * Logs a version refresh statement with the workflow's descriptor language if workflow is a FULL workflow .
+     * These logs will be monitored by CloudWatch and displayed on Grafana.
+     * @param workflow
+     * @param workflowVersion
+     */
+    private void logWorkflowVersionRefresh(final Workflow workflow, final String workflowVersion) {
+        if (workflow.getMode() == WorkflowMode.FULL) {
+            LOG.info(String.format("%s: Refreshing version %s for %s workflow named %s", dashboardPrefix, workflowVersion, workflow.getDescriptorType(), workflow.getEntryPath()));
+        }
+    }
+
     @GET
     @Path("/{workflowId}/refresh")
     @Timed
     @UnitOfWork
+    @Operation(operationId = "refresh", description = "Refresh one particular workflow.", security = @SecurityRequirement(name = "bearer"))
     @ApiOperation(nickname = "refresh", value = "Refresh one particular workflow.", notes = "Full refresh", authorizations = {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Workflow.class)
     public Workflow refresh(@ApiParam(hidden = true) @Auth User user,
         @ApiParam(value = "workflow ID", required = true) @PathParam("workflowId") Long workflowId) {
-        Workflow workflow = workflowDAO.findById(workflowId);
-        checkEntry(workflow);
-        checkUser(user, workflow);
-        checkNotHosted(workflow);
+        return refreshWorkflow(user, workflowId, Optional.empty());
+    }
+
+    @GET
+    @Path("/{workflowId}/refresh/{version}")
```

Comment:
Paging @garyluu 
Does a shared prefix with `@Path("/{workflowId}/refresh")` cause issues with ELK?

Reply:
No, there's a slash, it should be fine.  but as always, the script to generate the patterns should always be ran.